lite/base_certifier_test.go

```diff
@@ -1,59 +1,58 @@
-package lite_test
+package lite
 
 import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 
+	lerr "github.com/tendermint/tendermint/lite/errors"
 	"github.com/tendermint/tendermint/types"
-
-	"github.com/tendermint/tendermint/lite"
-	liteErr "github.com/tendermint/tendermint/lite/errors"
 )
 
-func TestStaticCert(t *testing.T) {
+func TestBaseCert(t *testing.T) {
 	// assert, require := assert.New(t), require.New(t)
 	assert := assert.New(t)
 	// require := require.New(t)
 
-	keys := lite.GenValKeys(4)
+	keys := genPrivKeys(4)
 	// 20, 30, 40, 50 - the first 3 don't have 2/3, the last 3 do!
 	vals := keys.ToValidators(20, 10)
 	// and a certifier based on our known set
 	chainID := "test-static"
-	cert := lite.NewStaticCertifier(chainID, vals)
+	cert := NewBaseCertifier(chainID, 2, vals)
 
 	cases := []struct {
-		keys        lite.ValKeys
+		keys        privKeys
 		vals        *types.ValidatorSet
 		height      int64
 		first, last int  // who actually signs
 		proper      bool // true -> expect no error
 		changed     bool // true -> expect validator change error
 	}{
+		// height regression
+		{keys, vals, 1, 0, len(keys), false, false},
 		// perfect, signed by everyone
-		{keys, vals, 1, 0, len(keys), true, false},
+		{keys, vals, 2, 0, len(keys), true, false},
 		// skip little guy is okay
-		{keys, vals, 2, 1, len(keys), true, false},
+		{keys, vals, 3, 1, len(keys), true, false},
 		// but not the big guy
-		{keys, vals, 3, 0, len(keys) - 1, false, false},
-		// even changing the power a little bit breaks the static validator
-		// the sigs are enough, but the validator hash is unknown
-		{keys, keys.ToValidators(20, 11), 4, 0, len(keys), false, true},
+		{keys, vals, 4, 0, len(keys) - 1, false, false},
+		// Changing the power a little bit breaks the static validator.
+		// The sigs are enough, but the validator hash is unknown.
+		{keys, keys.ToValidators(20, 11), 5, 0, len(keys), false, true},
 	}
 
 	for _, tc := range cases {
-		check := tc.keys.GenCommit(chainID, tc.height, nil, tc.vals,
+		sh := tc.keys.GenSignedHeader(chainID, tc.height, nil, tc.vals, tc.vals,
```

Comment:
Test can probably be simplified and make more explicit to have a valSet and few cases for different signedHeaders, one with enough voting power, one with less and one where valSet does not match.

Reply:
I agree, but arguably out of scope for this PR.  Please make an issue if you think necessary?  I think it's OK, but maybe worth discussing.